testutil/fuzz/image_fuzz.go

```diff
@@ -0,0 +1,44 @@
+// +build gofuzz
+
+package fuzz
+
+import (
+	"github.com/docker/docker/image"
+	"github.com/docker/docker/layer"
+	"io/ioutil"
+	"os"
+	"runtime"
+)
+
+type mockLayerGetReleaser struct{}
+
+func (ls *mockLayerGetReleaser) Get(layer.ChainID) (layer.Layer, error) {
+	return nil, nil
+}
+
+func (ls *mockLayerGetReleaser) Release(layer.Layer) ([]layer.Metadata, error) {
+	return nil, nil
+}
+
+func FuzzImage(data []byte) int {
+	tmpdir, err := ioutil.TempDir("", "images-fs-store")
+	defer os.RemoveAll(tmpdir)
+	if err != nil {
+		return -1
+	}
+	fsBackend, err := image.NewFSStoreBackend(tmpdir)
+	if err != nil {
+		return -1
+	}
+	mlgrMap := make(map[string]image.LayerGetReleaser)
+	mlgrMap[runtime.GOOS] = &mockLayerGetReleaser{}
+	store, err := image.NewImageStore(fsBackend, mlgrMap)
+	if err != nil {
+		return 0
+	}
+	_, err = store.Create(data)
```

Comment:
I'm not sure this function is worth fuzzing. Can we rather fuzz Dockerfile parser and Docker Registry client packages?

Reply:
Are you suggesting we reopen https://github.com/moby/buildkit/pull/1518?

Reply:
Yes, that would be appreciated

Reply:
Hereby reopened at https://github.com/moby/buildkit/pull/1813, so I am marking this one as resolved